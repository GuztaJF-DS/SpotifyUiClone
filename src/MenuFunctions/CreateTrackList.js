import React from "react";
import { PlaySong } from "../Redux/SongSlice";

export default function CreateTrackList(props){

    let Tracks=props.JSONData[props.AlbumId].TrackList;
    let Cover=props.JSONData[props.AlbumId].Cover
    let dispatch=props.dispatch
    let TracksJsx=[];

    Tracks.map(function(data,index){
        TracksJsx.push(
                <button key={index} onClick={
                    ()=>{
                        dispatch(
                            PlaySong(
                                {"TrackList":Tracks,"SongId":index,"SongCover":Cover}
                            )
                        )
                    }
                }
                className="SongButton"
                >
                    <div className="SongName">
                    {data.Name}
                    </div>
                    <div className="SongDuration">
                    {data.Duration}
                    </div>
                </button>
            )
        })
        
        return TracksJsx
}