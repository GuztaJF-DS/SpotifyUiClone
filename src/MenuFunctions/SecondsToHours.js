export default function SecondsToHours(seg) {
    let Hours=Math.floor(seg/(60*60));
    let Minutes=Math.floor(seg/60);
    if(Minutes>=60){
        Minutes=Minutes-60;
    }

    var leftover=seg%(60*60);
    leftover%=60;
    
    let Seconds=Math.ceil(leftover)
    
    let final=(Hours===0)
    ?Minutes+":"+Seconds
    :Hours+":"+Minutes+":"+Seconds


    return final
}