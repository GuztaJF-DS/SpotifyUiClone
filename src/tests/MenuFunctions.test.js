import SecondsToHours from '../MenuFunctions/SecondsToHours';
import CreateTrackList from '../MenuFunctions/CreateTrackList';
import TestData from './Testdata.json';
import renderer from 'react-test-renderer'

test('it Returns the correct Hour',()=>{
    expect(SecondsToHours('4719')).toBe('1:18:39')
}) 

test("It Creates TrackList",()=>{
    const tree=renderer.create(<CreateTrackList JSONData={TestData} AlbumId={'0'}/>)
    expect(tree).toMatchSnapshot()
})
