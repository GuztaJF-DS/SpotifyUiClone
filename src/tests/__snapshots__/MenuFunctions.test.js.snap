// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It Creates TrackList 1`] = `
Array [
  <button
    className="SongButton"
    onClick={[Function]}
  >
    <div
      className="SongName"
    >
      Song 1
    </div>
    <div
      className="SongDuration"
    >
      2:57
    </div>
  </button>,
  <button
    className="SongButton"
    onClick={[Function]}
  >
    <div
      className="SongName"
    >
      Song 2
    </div>
    <div
      className="SongDuration"
    >
      2:10
    </div>
  </button>,
]
`;
